plugins/transfer/plugin.go

```diff
@@ -59,68 +59,111 @@ func init() {
 				return nil, err
 			}
 
-			d, err := ic.Get(plugin.DiffPlugin)
-			if err != nil {
-				return nil, err
-			}
-
 			// Set configuration based on default or user input
 			var lc local.TransferConfig
-			lc.MaxConcurrentDownloads = config.maxConcurrentDownloads
-			lc.MaxConcurrentUploadedLayers = config.maxConcurrentUploadedLayers
-			for _, uc := range config.unpackConfiguration {
-				p, err := platforms.Parse(uc.platform)
+			lc.MaxConcurrentDownloads = config.MaxConcurrentDownloads
+			lc.MaxConcurrentUploadedLayers = config.MaxConcurrentUploadedLayers
+			for _, uc := range config.UnpackConfiguration {
+				p, err := platforms.Parse(uc.Platform)
 				if err != nil {
-					return nil, fmt.Errorf("%s: platform configuration %v invalid", plugin.TransferPlugin, uc.platform)
+					return nil, fmt.Errorf("%s: platform configuration %v invalid", plugin.TransferPlugin, uc.Platform)
+				}
+
+				sn := ms.Snapshotter(uc.Snapshotter)
```

Comment:
do we need to check supported platform for snapshot here?

Reply:
I don't think so, the configuration already defines the platform to match against. The snapshotter platforms are a little weird because they more often represent the system which containerd is running on, which we also enforce through compile options. Really though, the snapshotters are normally ok for any architecture on the same OS and likely even across some OSes that it may not be explicitly configured for.